api/devicerepository.proto

```diff
@@ -308,6 +312,20 @@ message MessagePayloadFormatter {
   PayloadFormatter formatter = 1 [(validate.rules).enum.defined_only = true];
   // Parameter for the formatter, must be set together.
   string formatter_parameter = 2;
+
+  string codec_id = 3 [(gogoproto.customname) = "CodecID", (validate.rules).string = {pattern: "^([a-z0-9](?:[-]?[a-z0-9]){2,}|)?$" , max_len: 36}];
+
+  message Example {
+    string description = 1;
+    message Input {
+      uint32 f_port = 1 [(validate.rules).uint32.lte = 255];
+      bytes frm_payload = 2 [(gogoproto.customname) = "FRMPayload"];
+    };
```

Comment:
This is the input for decoding uplink and decoding downlink. How about encoding downlink?